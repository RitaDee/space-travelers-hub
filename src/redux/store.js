import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/MissionSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
