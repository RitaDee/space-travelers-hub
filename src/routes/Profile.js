import React from 'react';
import { useSelector } from 'react-redux';
import { filterReservedRockets } from '../redux/rockets/rocketsSlice';
import styles from '../styles/Profile.module.css';

const Profile = () => {
  const reservedRockets = useSelector(filterReservedRockets);

  return (
    <main className={styles.main}>
      <section className={styles.section}>
        <h2 className={styles.title}>My Rockets</h2>
        <ul className={styles.list}>
          {reservedRockets.map((e) => (
            <li key={e.rocket_id} className={styles.listItem}>
              {e.rocket_name}
            </li>
          ))}
        </ul>
      </section>
    </main>
  );
};

export default Profile;
