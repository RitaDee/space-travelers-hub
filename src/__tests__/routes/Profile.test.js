import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Profile from '../../routes/Profile';

describe('Profile', () => {
  it('renders correctly for reserved state', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
