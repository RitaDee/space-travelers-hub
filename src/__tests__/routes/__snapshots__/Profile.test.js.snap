// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly for reserved state 1`] = `
<main
  className="main"
>
  <section
    className="section"
  >
    <h2
      className="title"
    >
      My Missions
    </h2>
    <ul
      className="list"
    />
  </section>
  <section
    className="section"
  >
    <h2
      className="title"
    >
      My Rockets
    </h2>
    <ul
      className="list"
    />
  </section>
</main>
`;
